libraries/botbuilder-dialogs/src/dialog.ts

```diff
@@ -417,4 +417,63 @@ export abstract class Dialog<O extends object = {}> extends Configurable {
 
         return label;
     }
+
+    protected async endParentDialog(dc: DialogContext, result?: any): Promise<DialogTurnResult> {
+        this.popCommands(dc);
+        if (dc.stack.length > 0 || !dc.parent) {
+            return await dc.endDialog(result);
+        } else {
+            const turnResult = await dc.parent.endDialog(result);
+            turnResult.parentEnded = true;
+            return turnResult;
+         }
+    }
+
+    protected async replaceParentDialog(dc: DialogContext, dialogId: string, options?: object): Promise<DialogTurnResult> {
+        this.popCommands(dc);
+        if (dc.stack.length > 0 || !dc.parent) {
+            return await dc.replaceDialog(dialogId, options);
+        } else {
+            const turnResult = await dc.parent.replaceDialog(dialogId, options);
+            turnResult.parentEnded = true;
+            return turnResult;
+         }
+    }
+
+    protected async repeatParentDialog(dc: DialogContext, options?: object): Promise<DialogTurnResult> {
+        this.popCommands(dc);
+        if (dc.stack.length > 0 || !dc.parent) {
+            return await dc.replaceDialog(dc.activeDialog.id, options);
+        } else {
+            const turnsResult = await dc.parent.replaceDialog(dc.parent.activeDialog.id, options);
+            turnsResult.parentEnded = true;
+            return turnsResult;
+         }
+    }
+
+    protected async cancelAllParentDialogs(dc: DialogContext): Promise<DialogTurnResult> {
+        this.popCommands(dc);
+        if (dc.stack.length > 0 || !dc.parent) {
+            return await dc.cancelAllDialogs();
+        } else {
+           const turnResult = await dc.parent.cancelAllDialogs();
+           turnResult.parentEnded = true;
+           return turnResult;
+        }
+    }
+
+    private popCommands(dc: DialogContext): void {
```

Comment:
>popCommands [](start = 12, length = 11)

This shouldn't be needed as we removed the concept of inherited memory.  This may be on the C# side of things but should be removed from there as well.